frappe/database/database.py

```diff
@@ -956,6 +957,120 @@ def set_value(
+		"""
+		:param doctype: DocType to update
+		:param docs: List of dictionaries to update
+		:param chunk_size: Number of documents to update in a single transaction
+		:param commit: Commit after each chunk
+
+		docs should be in the following format:
+		```py
+		[
+		    {
+		        name: "name1",
+		        field1: "value1",
+		        field2: "value2",
+		    },
+		    {
+		        name: "name2",
+		        field1: "value1",
+		        field2: "value2",
+		    },
+		    ...,
+		]
+		```
+
+		Note:
+		    - Each dictionary should have a key `name` which is the docname of the document to update.
+		    - The rest of the keys are the fields to update.
+		    - The fields to update should be the same for all documents.
+		    - Bigger chunk sizes could be less performant. Use appropriate chunk size based on the number of fields to update.
+
+		"""
+		if not docs:
+			return
+
+		total_docs = len(docs)
+
+		for index in range(0, total_docs, chunk_size):
+			chunk = docs[index : index + chunk_size]
+			docs_map = {doc.pop("name"): doc for doc in chunk}
+
+			self.build_query_and_update_in_bulk(doctype, docs_map)
+
+			if commit:
+				frappe.db.commit()  # nosemgrep
+
+	@staticmethod
+	def build_query_and_update_in_bulk(doctype: str, docs: dict):
+		"""
+		:param doctype: DocType to update
+		:param docs: Dictionary of key (docname) and values to update (must < 100)
+
+		---
+
+		docs should be in the following format:
+		```py
+		{
+		    "docname1": {
+		        "field1": "value1",
+		        "field2": "value2",
+		        ...
+		    },
+		    "docname2": {
+		        "field1": "value1",
+		        "field2": "value2",
+		        ...
+		    },
+		}
+		```
+
+		---
+
+		Query will be built as:
+		```sql
+		UPDATE `tabItem`
+		SET `status` = CASE
+		    WHEN `name` = 'Item-1' THEN 'Close'
+		    WHEN `name` = 'Item-2' THEN 'Open'
+		    WHEN `name` = 'Item-3' THEN 'Close'
+		    WHEN `name` = 'Item-4' THEN 'Cancelled'
+		    ELSE NULL
+		end,
+		`description` = CASE
+		    WHEN `name` = 'Item-1' THEN 'This is the first task'
+		    WHEN `name` = 'Item-2' THEN 'This is the second task'
+		    WHEN `name` = 'Item-3' THEN 'This is the third task'
+		    WHEN `name` = 'Item-4' THEN 'This is the fourth task'
+		    ELSE NULL
+		end
+		WHERE  `name` IN ( 'Item-1', 'Item-2', 'Item-3', 'Item-4' )
+		```
+		"""
+		if not docs:
+			return
+
+		dt = frappe.qb.DocType(doctype)
+		update_query = frappe.qb.update(dt)
+
+		fields_to_update = next(iter(docs.values())).keys()
+		docnames = list(docs.keys())
+
+		# Initialize CASE
+		conditions = frappe._dict({field: Case() for field in fields_to_update})
+
+		# WHEN conditions
+		for docname, row in docs.items():
+			for field, value in row.items():
+				conditions[field].when(dt.name == docname, value)
+
+		# ELSE conditions
+		for field in conditions:
+			update_query = update_query.set(dt[field], conditions[field].else_(None))
```

Comment:
not needed since we don't allow heterogenous data anyway. also `ELSE NULL` can lead to data destruction